src/theme-provider.js

```diff
@@ -53,13 +58,12 @@ class ThemeProvider extends React.Component {
     // set broadcast state by merging outer theme with own
     if (this.context[CHANNEL]) {
       this.setOuterTheme(this.context[CHANNEL].getState())
-      this.broadcast.setState(this.getTheme())
     }
   }
 
   componentWillReceiveProps(nextProps) {
     if (this.props.theme !== nextProps.theme) {
-      this.broadcast.setState(this.getTheme(nextProps.theme))
+      this.publishTheme(nextProps.theme)
```

Comment:
Pardon my ignorance, but how has this made an effectual change? `publishTheme` is calling `broadcast.setState(this.getTheme(nextProps.theme))` which is was was here before. It's only been moved to a separate method.

Reply:
oh, this part is actually not a fix of any sorts, i just have extracted it to separate method so it would be easier to reuse it in `setOuterTheme`

Reply:
OK, thanks for clarifying.